pkg/webhook/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -65,6 +86,40 @@ func checkRevisionName(appConfig *v1alpha2.ApplicationConfiguration) (bool, stri
 	return true, ""
 }
 
+// checkWorkloadNameForVersioning check whether versioning-enabled component workload name is empty
+func checkWorkloadNameForVersioning(ctx context.Context, client client.Reader, appsClient clientappv1.ControllerRevisionsGetter, appConfig *v1alpha2.ApplicationConfiguration) (bool, string) {
```

Comment:
I strongly recommend testing this individually instead of combining it with the handler test. That is not a good unit test pattern. That test is already way too big, not readable, maintainable.